.travis.yml

```diff
@@ -99,68 +99,108 @@ geosx_pangea_build: &geosx_pangea_build
   - gcloud auth activate-service-account --key-file=${GEOSX_GCLOUD_KEY}
   - gsutil cp -a public-read ${GEOSX_BUNDLE} gs://geosx/Pangea2/
 
+draft_script: &draft_script
+  script:
+  # TRAVIS_PULL_REQUEST is false if job is not from a PR
+  - if [[ $TRAVIS_PULL_REQUEST == false ]]; then exit 0; fi;
+  - |
+    is_draft=$(curl -H "Accept: application/vnd.github.v3+json" \
+        https://api.github.com/repos/GEOSX/GEOSX/pulls/$TRAVIS_PULL_REQUEST | \
+        jq ".draft")
+
+  # CI jobs will be cancelled if PR is a draft.
+  # PR status must be "Open" to run CI.
+  - |
+    if [[ $is_draft == true ]]; then
+      exit 1
```

Comment:
Also, doing this will mark the job as `failed` and not `cancelled`.
I think that cancelling a `build` or a `job` requires a web service call, and thus a travis-ci secret token (even if we are open now, I don't think anyone can cancel one of our jobs). Maybe when you are inside of a travis VM you need no token?
I'm OK with this since it simplifies the code, but the message to the user is more blurry.

Reply:
@TotoGaz  Yes, it looks like an access token is still required when running inside the VM. I was able to generate an access token, but I'm not sure how to keep it hidden in the `.travis.yml`.

Reply:
It is possible to store it as a _secret_ environment variable in the travis-ci console.

@rrsettgast Do you think it is worth the price? Or are you OK with the job being `errored`?

Reply:
We can store an env variable. I don't think there is any cost/drawback...is there?

Reply:
I mean no cost in terms of CPUs.

We just add another one REST call + having credentials to look at for little benefit: simple `exit 1` does the main job.
Going further is polishing and adding more code.

Reply:
I did some more digging, found this Travis tutorial [here](https://docs.travis-ci.com/user/encryption-keys/) to add an encrypted environment variable through the Travis command line interface. Seems to be working, shows the build status as "Cancelled" instead of failed.